spec/models/state_changer/mx_tube_spec.rb

```diff
@@ -15,11 +15,21 @@
 
   let(:user) { build_stubbed :user }
   let(:customer_accepts_responsibility) { false }
-  let(:labware) { create :tube }
-  let!(:transfer_request) { create :transfer_request, target_asset: labware.receptacle, state: transfer_request_state }
-  let!(:request) { create :request, target_asset: labware.receptacle, state: request_state }
```

Comment:
Yes like this, probably it was creating problems using let!() :) sorry the previous comment then

Reply:
OK yes I forgot about this, I think the reason I changed it here was because I had some problem with the order of execution when nesting contexts within contexts.